### PR TITLE
chore(reports): change default executor to owner

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
This changes Alerts & Reports to execute as the primary owner of the alert/report by default. In cases where there are multiple owners, the last modifier and creator are given priority in this order, otherwise the first owner will be used. This option was introduced in #21931.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
@villebro This is great, and we're looking forward to testing it out. I assume that this is a 3.0 chore, and we should merge this when the branch opens up for breaking changes, but if someone wants to test out the functionality now, they would only need to change this config? `ALERT_REPORTS_EXECUTE_AS: List[ExecutorType] = [ExecutorType.OWNER]`

**Comment by villebro** [comment]
> @villebro This is great, and we're looking forward to testing it out. I assume that this is a 3.0 chore, and we should merge this when the branch opens up for breaking changes, but if someone wants to test out the functionality now, they would only need to change this config? `ALERT_REPORTS_EXECUTE_AS: List[ExecutorType] = [ExecutorType.OWNER]`

@eschutho Yes, I think it's a good idea to slot this for 3.0. I'll tag it accordingly. And by changing the config as as proposed above you can opt in to this behavior.

**Comment by rusackas** [comment]
If this is considered breaking, we should put it on the 3.0 project board and seek lazy consensus. The merge window is open!

**Review by eschutho — APPROVED** [review]

File: UPDATING.md [hunk 1]
```diff
@@ -50,8 +50,9 @@ assists people when migrating to a new version.
 
 ### Breaking Changes
 
-- [23651](https://github.com/apache/superset/pull/23651) Removes UX_BETA feature flag.
-- [23663](https://github.com/apache/superset/pull/23663) Removes deprecated feature flags `ALLOW_DASHBOARD_DOMAIN_SHARDING`, `DISPLAY_MARKDOWN_HTML`, and `FORCE_DATABASE_CONNECTIONS_SSL`.
+- [22799](https://github.com/apache/superset/pull/22799): Alerts & Reports has been changed to execute as the owner of the alert/report by default, giving priority to the last modifier and then the creator if either is contained within the list of owners, otherwise the first owner will be used. To continue using the selenium user, please add the following to your `superset_config.py`: `ALERT_REPORTS_EXECUTE_AS = ["selenium"]`
+- [23651](https://github.com/apache/superset/pull/23651): Removes UX_BETA feature flag.
+- [23663](https://github.com/apache/superset/pull/23663): Removes deprecated feature flags `ALLOW_DASHBOARD_DOMAIN_SHARDING`, `DISPLAY_MARKDOWN_HTML`, and `FORCE_DATABASE_CONNECTIONS_SSL`.
```
Comment (villebro) [inline UPDATING.md:55]: bycatch: the colon was missing after the link

**Pull request merged**